app/lib/tool/utils/emoji.dart

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Removes emojis from [text].
+///
+/// TODO: consider using a library that provides support for doing the same.
+String removeEmojis(String text) {
+  // shortcut for simple text
+  if (text.runes.every((r) => r < 256)) return text;
+
+  final runes = text.runes.toList();
+  final newRunes = <int>[];
+  for (var i = 0; i < runes.length; i++) {
+    final c = runes[i];
+    if (c > 256 && _basicEmojiRanges.any((r) => r.start <= c && c < r.end)) {
+      continue;
+    }
+    newRunes.add(c);
+  }
+  return String.fromCharCodes(newRunes);
+}
+
+/// Ranges were generated by the `_UnicodeEmojiSequences.getBasicRanges`
+/// method, with a merge threshold of 8. It collapses the ranges with a
+/// reasonable rate where we may include characters that are not necessarily
+/// emojies.
+///
+/// print(sequences.getBasicRanges(mergeThreshold: 8).map((e) => '$e\n').join());
```

Comment:
How bad is mergeThreshold: 0 ?
I guess this is not a very performance-sensitive operation...